### PR TITLE
Stdev dcmorales t78 make a loader

### PR DESCRIPTION
**Pull request by stdevDcmorales**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -42,3 +42,8 @@ Thumbs.db
 
 # compiled binaries
 /explorer.*
+/dist/vendor.181584baefd2a47c5eda.bundle.js
```
Comment (olemis) [inline .gitignore]: @stdevDcmorales you should add a generic rule here e.g. `dist/*` or `dist/*.bundle.*` .

File: .gitignore [hunk 2]
```diff
@@ -42,3 +42,65 @@ Thumbs.db
...
+/dist/favicon.ico
+/dist/index.html
+/dist/inline.552abd77f383c2cb434b.bundle.js
+/dist/main.0bb75a91f865ef2b3d8a.bundle.js
+/dist/polyfills.35726d60cdf25fecc5f1.bundle.js
+/dist/scripts.7526cd49a5fb58ce73f1.bundle.js
+/dist/skycoin.ico
+/dist/styles.ad5f424b95bd73aa89ae.bundle.css
+/dist/vendor.1c15142c9c6efd45619a.bundle.js
+/node_modules/angular-font-awesome
```
Comment (olemis) [inline .gitignore]: Same again .. just ignore `node_modules/*` as a whole .

**Comment by gz-c** [comment]
We commit dist to the repo so that we dont have to install nodejs to deploy

File: .angular-cli.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
       "prefix": "app",
       "styles": [
         "assets/bootstrap.min.css",
+        "../node_modules/font-awesome/css/font-awesome.css",
```
Comment (gz-c) [inline .angular-cli.json]: doesnt this need to be in assets?

**Comment by samuelvisscher** [comment]
Sorry for possible conflicting messages, but please leave old dist in place without generating new dist. Dist files should be compiled by core members (steve, me, iketheadore, synth)

**Comment by olemis** [comment]
@stdevDcmorales I see some merge conflicts in this branch .

File: .angular-cli.json (hunk 3 above)
Comment (samuelvisscher) [reply .angular-cli.json]: Yes, copy to assets and use a minimized lib

File: .gitignore [hunk 4]
```diff
@@ -42,3 +42,4 @@ Thumbs.db
```
Comment (samuelvisscher) [inline .gitignore]: Don't add dist to gitignore

Comment (samuelvisscher) [reply .gitignore]: To take away npm dependency on our servers

File: dist/3rdpartylicenses.txt [hunk 5]
```diff
@@ -1,92 +0,0 @@
-core-js@2.5.1
```
Comment (samuelvisscher) [inline dist/3rdpartylicenses.txt]: Don't rebuild dist files, Steve or I will generate dist files. This makes merging, reviewing and ensuring code integrity easier.

File: package.json [hunk 6]
```diff
@@ -22,7 +22,9 @@
     "@angular/platform-browser": "^4.0.0",
     "@angular/platform-browser-dynamic": "^4.0.0",
     "@angular/router": "^4.0.0",
+    "angular-font-awesome": "^3.1.1",
```
Comment (samuelvisscher) [inline package.json]: No need to use an angular package for something as trivial as fa

File: package.json [hunk 7]
```diff
@@ -22,7 +22,9 @@
...
     "core-js": "^2.4.1",
+    "font-awesome": "^4.7.0",
```
Comment (samuelvisscher) [inline package.json]: just download the min.css

File: src/app/app.module.ts [hunk 8]
```diff
@@ -9,14 +9,13 @@ import { SearchBarComponent } from './components/layout/search-bar/search-bar.co
...
 import { ApiService } from './services/api/api.service';
-import { HttpModule } from '@angular/http';
 import { BlockDetailsComponent } from './components/pages/block-details/block-details.component';
 import { TransactionDetailComponent } from './components/pages/transaction-detail/transaction-detail.component';
 import { AddressDetailComponent } from './components/pages/address-detail/address-detail.component';
 import { TransactionsValuePipe } from './pipes/transactions-value.pipe';
 import { ExplorerService } from './services/explorer/explorer.service';
 import { QrCodeComponent } from './components/layout/qr-code/qr-code.component';
-import { FormsModule } from '@angular/forms';
+import { SharedModule } from "app/modules/shared/shared.module";
```
Comment (samuelvisscher) [inline src/app/app.module.ts]: Although using a SharedModule is quite common, due to the size of the application we have chosen not to do so

File: src/app/components/pages/blocks/blocks.component.spec.ts [hunk 9]
```diff
@@ -1,20 +1,20 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
-import { BlockChainTableComponent } from './block-chain-table.component';
+import { BlocksComponent } from './blocks.component';
 
-describe('BlockChainTableComponent', () => {
-  let component: BlockChainTableComponent;
-  let fixture: ComponentFixture<BlockChainTableComponent>;
+describe('BlocksComponent', () => {
+    let component: BlocksComponent;
```
Comment (samuelvisscher) [inline src/app/components/pages/blocks/blocks.component.spec.ts]: We use 2 spaces for indentation

File: src/app/components/pages/blocks/blocks.component.spec.ts [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
+  let fixture: ComponentFixture<BlocksComponent>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ BlockChainTableComponent ]
+        declarations: [BlocksComponent ]
```
Comment (samuelvisscher) [inline src/app/components/pages/blocks/blocks.component.spec.ts]: Please maintain spaces between square brackets

File: src/app/modules/shared/components/loading/loading.component.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'loading',
+  templateUrl: './loading.component.html',
+  styleUrls: ['./loading.component.scss']
+})
+export class LoadingComponent implements OnInit {
+    
+    @Input() name: string;
```
Comment (samuelvisscher) [inline src/app/modules/shared/components/loading/loading.component.ts]: This is really overengineered for what we're doing

File: src/index.html [hunk 12]
```diff
@@ -11,6 +11,10 @@
   <link rel="icon" type="image/x-icon" href="skycoin.ico">
 </head>
 <body>
-  <app-root></app-root>
+  <app-root>
```
Comment (samuelvisscher) [inline src/index.html]: The spinner was intended for inside the table

Comment (samuelvisscher) [reply src/index.html]: And on loading a new address

**Comment by samuelvisscher** [comment]
Will include this functionality in my next PR, so closing

**Pull request closed without merging**